### PR TITLE
cmd/version: print os/version, kernel and bitness

### PR DESCRIPTION
**Pull request by ivandeex**
#### What is the purpose of this change?

Make `rclone version`  print runtime OS version, kernel and architecture bitness.

The implementation is based on https://github.com/shirou/gopsutil (namely [gopsutil/host](https://github.com/shirou/gopsutil/tree/master/host))
The detection on OpenBSD is limited to OS name because base package fails to build on it.

#### Was the change discussed in an issue or in the forum before?

Related to https://github.com/rclone/rclone/issues/5121#issuecomment-803612181

References:
- https://unix.stackexchange.com/questions/92199/how-can-i-reliably-get-the-operating-systems-name
- https://www.reddit.com/r/golang/comments/f6wrtv/how_to_determine_32_or_64_bit_on_host_machine/
- https://stackoverflow.com/questions/7011071/detect-32-bit-or-64-bit-of-windows
- https://github.com/golang/go/issues/24843
- https://github.com/zcalusic/sysinfo#sample-output
- https://github.com/shirou/gopsutil/tree/v3.21.3/host

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [ ] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

#### Testing

Instead of writing a unit test which is rather useless in this case, I will publish
beta releases for linux/windows tagged with `v1.55.0-osversionNN`.
See https://github.com/ivandeex/rclone/releases
Beta testers can download a beta and decide whether `rclone version` output feels correct or not.

### CODE REVIEW

**Comment by ivandeex** [comment]
`v1.55.0-osversion01` prints for me

on Linux:
```
- os/version: ubuntu 18.04 (64 bit)
- os/kernel: 4.15.0-136-generic (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.13.4
...
```

on Windows:
```
- os/version: Microsoft Windows 8.1 Pro 6.3.9600 Build 9600 (64 bit)
- os/kernel: 6.3.9600 Build 9600 (x86_64)
- os/type: windows
- os/arch: amd64
- go/version: go1.13.4
- go/linking: dynamic
- go/tags: cmount
```

@ncw @albertony @darthShadow
Your comments are welcome

**Comment by albertony** [comment]
Great, useful with all those details!
A bit long os/version string on Windows, and some overlap with the os/kernel string, but i suppose thats the way the third party lib returns it? Better too much than too little, anyway..

**Comment by ivandeex** [comment]
[osversion02](https://github.com/ivandeex/rclone/releases/tag/v1.55.0-osversion02)
Linux: same
Windows:
```
- os/version: Microsoft Windows 8.1 Pro (64 bit)
- os/kernel: 6.3.9600 Build 9600 (x86_64)
- os/type: windows
- os/arch: amd64
- go/version: go1.13.4
- go/linking: dynamic
- go/tags: cmount
```

**Comment by albertony** [comment]
Win10 computer:
```
rclone v1.55.0-osversion02
- os/version: Microsoft Windows 10 Enterprise (64 bit)
- os/kernel: 10.0.18363 Build 18363 (x86_64)
- os/type: windows
- os/arch: amd64
- go/version: go1.13.4
- go/linking: dynamic
- go/tags: cmount
```
Arch in WSL:
```
rclone v1.55.0-osversion02
- os/version: arch (64 bit)
- os/kernel: 5.4.72-microsoft-standard-WSL2 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.13.4
- go/linking: dynamic
- go/tags: cmount
```

Looks much better! Ship it! :)

PS:

Wonder why the version "name" is not included, though.. I mean like "1909", "2004", "20H2" etc.. Thats the names that are usually used to describe minimum requirements etc. I don't have the build numbers in my head, so usually have to look them up in [wikipedia](https://en.wikipedia.org/wiki/Windows_10_version_history).

Also the version number in os/kernel includes the build number twice and does not include the patch version: "10.0.18363 Build 18363"..

From About section in settings panel:

image

From `winver`:

image

Nothing of this is important, though, I'm just wondering.. May have to have a look into what the gopsutil does some day, if nothing else just to satisfy my curiosity.

**Comment by ivandeex** [comment]
[v1.55.0-osversion03](https://github.com/ivandeex/rclone/releases/tag/v1.55.0-osversion03):

Windows 8:
```
- os/version: Microsoft Windows 8.1 Pro (64 bit)
- os/kernel: 6.3.9600 (x86_64)
- os/type: windows
- os/arch: amd64
- go/version: go1.13.4
- go/linking: dynamic
- go/tags: cmount
```

Linux - the same:
```
rclone v1.55.0-osversion03
- os/version: ubuntu 18.04 (64 bit)
- os/kernel: 4.15.0-136-generic (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.13.4
- go/linking: dynamic
- go/tags: cmount
```

> Wonder why the version "name" is not included, though.. I mean like "1909", "2004", "20H2" etc..

Also for linux I would expect `os/version: ubuntu 18.04 (bionic)`. I think it's a limitation of `gopsutil`, though it's not critical.

> what the gopsutil does

I guess on Windows it potters around registry, on Linux - in `/etc/xxx_release` and `uname -a`

**Comment by albertony** [comment]
Quick retest: Windows 10 output is similar to your Windows 8 output, WSL identical to before:
```
rclone v1.55.0-osversion03
- os/version: Microsoft Windows 10 Enterprise (64 bit)
- os/kernel: 10.0.18363 (x86_64)
- os/type: windows
- os/arch: amd64
- go/version: go1.13.4
- go/linking: dynamic
- go/tags: cmount
```

File: lib/buildinfo/osversion_openbsd.go
Comment (ncw) [inline lib/buildinfo/osversion_openbsd.go:12]: You could call [syscall.Uname](https://pkg.go.dev/golang.org/x/sys/unix#Uname) here and return [some things out of there](https://man7.org/linux/man-pages/man2/uname.2.html).

This should work on OpenBSD I think...

**Review by ncw — COMMENTED** [review]
I think this is looking great!

I put an idea for OpenBSD inline

File: lib/buildinfo/osversion_openbsd.go
Comment (ivandeex) [reply lib/buildinfo/osversion_openbsd.go:12]: The catch is how to test it. I don't have openbsd within reach.

Comment (ivandeex) [reply lib/buildinfo/osversion_openbsd.go:12]: We can try with virtualbox and https://app.vagrantup.com/generic/boxes/openbsd6 but there is no guarantee that resulting code doesn't panic with other actual flavors of OpenBSD.

Comment (ivandeex) [reply lib/buildinfo/osversion_openbsd.go:12]: I think I can open an upstream issue asking to make gopsutil work on OpenBSD. This OS is not primary for rclone. We can wait.

**Comment by ivandeex** [comment]
It would be interesting to look at MacOS output.

File: lib/buildinfo/osversion_openbsd.go
Comment (ivandeex) [reply lib/buildinfo/osversion_openbsd.go:12]: They already have a PR https://github.com/shirou/gopsutil/pull/993 and it's actively moving on. I've subscribed on it. We can either put this one on hold waiting for upstream or postpone dependency upgrade to a later PR.

**Comment by albertony** [comment]
The Windows 10 "version friendly-name" is easy enough to find from registry key:
`HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion`, value `ReleaseId`. On Windows 8 it does not exist, though ([ref](https://ss64.com/nt/ver.html)). Patch number is also there as registry value `UBR` (Update Build Revision), not sure if that is Windows 10 specific also.

So instead of:
>\- os/version: Microsoft Windows 10 Enterprise (64 bit)
>\- os/kernel: 10.0.18363 (x86_64)

We could have gotten something like:
> \- os/version: Microsoft Windows 10 Enterprise 1909 (64 bit)
> \- os/kernel: 10.0.18363.1440 (x86_64)

But then this requires Windows-specific registry access (if only a few lines), which I know you are not overly exited about :wink:, and for me also would just be "nice to have" at this point, so I'm fine with it as is. 

Regarding:
> Also the version number in os/kernel includes the build number twice and does not include the patch version: "10.0.18363 Build 18363"..

Just checked the gopsutil src, and it is definitely returning the same `dwBuildNumber` value twice:
```
version = fmt.Sprintf("%d.%d.%d Build %d", osInfo.dwMajorVersion, osInfo.dwMinorVersion, osInfo.dwBuildNumber, osInfo.dwBuildNumber)
```

So your fix should be safe, and always find match:

```
// Simplify kernel output: `RELEASE.BUILD Build BUILD` -> `RELEASE.BUILD`
match := regexp.MustCompile(`^([\d\.]+?\.)(\d+) Build (\d+)$`).FindStringSubmatch(osKernel)
if len(match) == 4 && match[2] == match[3] {
```

**Comment by ivandeex** [comment]
[v1.55.0-osversion04](https://github.com/ivandeex/rclone/releases/tag/v1.55.0-osversion04) includes `ReleaseId` and `Update Revision`

my build box:
```
rclone v1.55.0-osversion04
- os/version: ubuntu 18.04 (64 bit)
- os/kernel: 4.15.0-136-generic (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.13.4
- go/linking: dynamic
- go/tags: cmount
```

my desktop (includes update revision now):
```
rclone v1.55.0-osversion04
- os/version: Microsoft Windows 8.1 Pro (64 bit)
- os/kernel: 6.3.9600.18543 (x86_64)
- os/type: windows
- os/arch: amd64
- go/version: go1.13.4
- go/linking: dynamic
- go/tags: cmount
```

github runners (https://github.com/ivandeex/rclone/actions/runs/718936549):

ubuntu-latest:
```
rclone v1.55.0-osversion04
- os/version: ubuntu 20.04 (64 bit)
- os/kernel: 5.4.0-1043-azure (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.16.2
- go/linking: dynamic
- go/tags: cmount
```

windows_amd64 (confirmed having friendly name and update revision):
```
rclone v1.55.0-osversion04
- os/version: Microsoft Windows Server 2019 Datacenter 1809 (64 bit)
- os/kernel: 10.0.17763.1817 (x86_64)
- os/type: windows
- os/arch: amd64
- go/version: go1.16.2
- go/linking: dynamic
- go/tags: cmount
```

windows_i386 (notice that `os/arch` and `os/kernel arch` differ, it's allright)
```
rclone v1.55.0-osversion04
- os/version: Microsoft Windows Server 2019 Datacenter 1809 (64 bit)
- os/kernel: 10.0.17763.1817 (x86_64)
- os/type: windows
- os/arch: amd64
- go/version: go1.16.2
- go/linking: dynamic
- go/tags: cmount
```

mac_amd64 (macos-latest)
```
rclone v1.55.0-osversion04
- os/version: darwin 10.15.7 (64 bit)
- os/kernel: 19.6.0 (x86_64)
- os/type: darwin
- os/arch: amd64
- go/version: go1.16.2
- go/linking: dynamic
- go/tags: cmount
```

mac_ARM64 (something unexpected here, at least for me; github setup job shows `Mac OS X 10.15.7 19H524`, BUILD_FLAGS is `BUILD_FLAGS: -include "^darwin/arm64" -cgo -macos-arch arm64 -macos-sdk macosx11.1 -cgo-cflags=-I/usr/local/include -cgo-ldflags=-L/usr/local/lib`)
```
rclone v1.55.0-osversion04
- os/version: darwin 10.15.7 (64 bit)
- os/kernel: 19.6.0 (x86_64)
- os/type: darwin
- os/arch: amd64
- go/version: go1.16.2
- go/linking: dynamic
- go/tags: cmount
```

**Comment by albertony** [comment]
> We could have gotten something like:
> >
> > \- os/version: Microsoft Windows 10 Enterprise 1909 (64 bit)
> > \- os/kernel: 10.0.18363.1440 (x86_64)

What an easter miracle - now we do:
```
rclone v1.55.0-osversion04
- os/version: Microsoft Windows 10 Enterprise 1909 (64 bit)
- os/kernel: 10.0.18363.1440 (x86_64)
- os/type: windows
- os/arch: amd64
- go/version: go1.13.4
- go/linking: dynamic
- go/tags: cmount
```

WSL as before:
```
rclone v1.55.0-osversion04
- os/version: arch (64 bit)
- os/kernel: 5.4.72-microsoft-standard-WSL2 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.13.4
- go/linking: dynamic
- go/tags: cmount
```

PS: Is your "windows_i386" example copy/paste error? Identical to your "windows_amd64" example..

**Comment by ivandeex** [comment]
> PS: Is your "windows_i386" example copy/paste error? Identical to your "windows_amd64" example..

Yes, it was copy/paste error. The original was here https://github.com/ivandeex/rclone/runs/2269099401?check_suite_focus=true

> Why is it "64 bit" and "x86_64" but "os/arch: 386" though?

Because `os/arch` shows `GOARCH`, a build setting (it's actually go/arch).
`64 bit` and `x86_64` are taken from runtime, controlled by github workflow file https://github.com/ivandeex/rclone/actions/runs/718936549/workflow
Compare with https://github.com/rclone/rclone/blob/master/.github/workflows/build.yml#L51
It uses `windows-latest` runner (which is 64 bit) both for `windows_amd64` and `windows_386` builds.
Windows x86_64 can emulate `386` transparently for a `386` binary so it runs successfully.

File: lib/buildinfo/osversion_openbsd.go
Comment (ivandeex) [reply lib/buildinfo/osversion_openbsd.go:12]: What do you think @ncw ?

Comment (ncw) [reply lib/buildinfo/osversion_openbsd.go:12]: That sounds fine - it is a minority platform (sorry FreeBSD users) so happy to wait until that PR is merged.

**Review by ncw — APPROVED** [review]
LGTM :-)

File: lib/buildinfo/osversion_openbsd.go
Comment (ivandeex) [reply lib/buildinfo/osversion_openbsd.go:12]: OK. Will submit another PR when https://github.com/shirou/gopsutil/pull/993 ticks for me.

**Comment by ivandeex** [comment]
cmd/version: print os/version, kernel and bitness (#5204)

Related to #5121

Note: OpenBSD is a stub yet. This will be fixed when upstream PR gets resolved:
https://github.com/shirou/gopsutil/pull/993

**Pull request merged**
